libraries/cbor/src/macros.rs

```diff
@@ -41,18 +41,30 @@ macro_rules! cbor_map_options {
     };
 
     ( $( $key:expr => $value:expr ),* ) => {
+        cbor_extend_map_options! (
```

Comment:
What's the purpose of this new macro?

Reply:
It permits to append elements to an existing map instead of creating one from scratch.

Reply:
I find the naming confusing. For example `Vec::extend_from_slice` takes a Vec by `&mut ref`, whereas this *copies* the initial map.
If a copy happens anyway, is this new macro necessary?
How about simply using existing macros/functions:

```rust
let mut map = cbor_map_options! { /* key-value pairs */ };
map.extend(initial_map.iter());
```

Reply:
Where did you see that this copies the map? This would be a bug. The intent was to move the initial value.

Reply:
Then, the calling convention should be `extend_cbor_map_options!{ &mut map, /* key-value pairs */}`, with an explicit `&mut`.

And I think `extend_cbor_map_options` is more readable than `cbor_extend_map_options`. The function "extends" a "cbor-map", rather than "cbor-extending" a "map".

Reply:
Done.